test/model/compose_test.dart

```diff
@@ -314,6 +314,28 @@ hello
         '#narrow/dm/1,2-dm/near/12345',
         '#narrow/pm-with/1,2-pm/near/12345');
     });
+
+    test('normalize links to always include a "/" after hostname', () {
+      void checkGeneratedLink({required String realmUrl, required String expectedLink}) {
+        final account = eg.selfAccount.copyWith(realmUrl: Uri.parse(realmUrl));
+        final store = eg.store(account: account);
+        check(narrowLink(store, const CombinedFeedNarrow()).toString())
+          .equals(expectedLink);
+      }
+
+      checkGeneratedLink(
+        realmUrl:     'http://chat.example.com',
+        expectedLink: 'http://chat.example.com/#narrow');
```

Comment:
nit: always best to make the code transparent about what it's doing, when that doesn't come at a high cost in concision:
```suggestion
      check(generatedLink('http://chat.example.com'))
        .equals(          'http://chat.example.com/#narrow');
```

(And here it actually improves concision, too).

Then further can rename `generatedLink` to something a bit more specific, like `narrowLinkFor`.